controllers/cluster_controller.go

```diff
@@ -357,6 +383,54 @@ func (r *ClusterReconciler) postClusterProviderReconcile(ctx context.Context, lo
 	return controller.Result{}, nil
 }
 
+func (r *ClusterReconciler) updateStatus(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) error {
+	// An EKS-A Cluster managed by the CLI is deleted after the cluster finalizer is removed, leaving the CAPI cluster behind. In this case, we do not want to update
+	// the status. If we do, patching will throw a 404 NotFound error when trying to update the status of the already deleted
+	// Cluster object.
+	if !cluster.DeletionTimestamp.IsZero() && metav1.HasAnnotation(cluster.ObjectMeta, anywherev1.ManagedByCLIAnnotation) {
+		log.Info("Cluster managed by CLI has been deleted but CAPI cluster may remain, skipping updating cluster status")
```

Comment:
If cluster is managed by CLI, why do we want to update status here? Are you trying to say Cluster managed by EKS-A, or what about the CLI makes it so that CAPI cluster remains and we want to update cluster status?

Reply:
Ah. I'm sorry for the confusion. 

What we're trying to say here is that when the cluster deletion process is being managed by the CLI, we **should not** update the status on the EKS-A cluster object because the reconciler will just remove the finalizer on the EKS-A cluster object during reconcileDelete [here](https://github.com/aws/eks-anywhere/blob/main/controllers/cluster_controller.go#L365). Therefore, we'd have no object to update the status of.

So, it's not that the CAPI cluster remains, but CLI is handling the deletion primarily. I'll update the comment and log here to be more clear.

Reply:
After talking offline that, we discussed that skipping updating the status when the cluster is marked for deletion AND has no finalizers is actually what we want to do here. The main thing is that we want to skip updating the status, if the cluster object has already been fully deleted because then subsequent patch operations would fail if the status is updated after it is fully deleted.